skydive_prof.go

```diff
@@ -28,33 +28,58 @@ import (
 	"log"
 	"os"
 	"os/signal"
+	"runtime"
 	"runtime/pprof"
 	"syscall"
 
 	"github.com/skydive-project/skydive/cmd/skydive"
 )
 
-func profile(c chan os.Signal) {
+func profile(cpu chan os.Signal, memory chan os.Signal) {
 	for {
-		<-c
-		f, err := os.Create("/tmp/skydive-cpu.prof")
-		if err != nil {
-			log.Fatal("could not create CPU profile: ", err)
-		}
-		if err := pprof.StartCPUProfile(f); err != nil {
-			log.Fatal("could not start CPU profile: ", err)
+		select {
+		case <-cpu:
+			f, err := os.Create("/tmp/skydive-cpu.prof")
+			if err != nil {
+				log.Fatal("could not create CPU profile: ", err)
+			}
+			if err := pprof.StartCPUProfile(f); err != nil {
+				log.Fatal("could not start CPU profile: ", err)
+			}
+			log.Println("CPU profiling Started")
+
+			<-cpu
+
+			pprof.StopCPUProfile()
+			log.Println("CPU profiling Stopped")
+
+			return
+		case <-memory:
+			// Memory Profile
+			runtime.GC()
+			memProfile, err := os.Create("/tmp/skydive-memory.prof")
+			if err != nil {
+				log.Fatal(err)
+			}
+			defer memProfile.Close()
```

Comment:
defer will never be executed, just call Close() explicitly after WriteHeapProfile()